gammagl/models/grace_pot.py

```diff
@@ -0,0 +1,283 @@
+from typing import Optional
+import torch
+import tensorlayerx as tlx
+from tensorlayerx import nn
+from torch.sparse import mm
+from gammagl.layers.conv import GCNConv
+from gammagl.utils import add_self_loops, calc_gcn_norm, degree, to_undirected, to_scipy_sparse_matrix
+import numpy as np
+import scipy.sparse as sp
+import pickle
+from scipy.sparse import coo_matrix
+import os.path as osp
+from time import perf_counter as t
+from gammagl.mpops import unsorted_segment_sum
+import tensorflow
+
+def BCEWithLogitsLoss(output, target):
+    return torch.nn.BCEWithLogitsLoss()(output, target)
+
+class LogReg(nn.Module):
+    def __init__(self, ft_in, nb_classes):
+        super(LogReg, self).__init__()
+        self.fc = nn.Linear(
+            in_features=ft_in, 
+            out_features=nb_classes, 
+            W_init=tlx.initializers.xavier_uniform(nb_classes)
+        )
+
+    def forward(self, seq):
+        ret = self.fc(seq)
+        ret = nn.LogSoftmax(dim=-1)(ret) 
+        return ret
+
+class Encoder(tlx.nn.Module):
+    def __init__(self, in_channels: int, out_channels: int, activation,
+                 base_model=GCNConv, k: int = 2):
+        super(Encoder, self).__init__()
+        self.base_model = base_model
+
+        assert k >= 2
+        self.k = k
+        self.conv = [base_model(in_channels, 2 * out_channels)]
+        for _ in range(1, k-1):
+            self.conv.append(base_model(2 * out_channels, 2 * out_channels))
+        self.conv.append(base_model(2 * out_channels, out_channels))
+        self.conv = nn.ModuleList(self.conv)
+
+        self.activation = activation
+
+    def forward(self, x:tlx.convert_to_tensor, edge_index:tlx.convert_to_tensor):
+        for i in range(self.k):
+            x = self.activation()(self.conv[i](x, edge_index))
+        return x
+
+
+class Model(tlx.nn.Module):
+    def __init__(self, encoder: Encoder, num_hidden: int, num_proj_hidden: int,
+                 tau: float = 0.5, dataset: str = "Cora"):
+        super(Model, self).__init__()
+        self.encoder: Encoder = encoder
+        self.tau: float = tau
+        self.dataset = dataset
+
+        self.fc1 = tlx.nn.Linear(in_features=num_hidden, out_features=num_proj_hidden)
+        self.fc2 = tlx.nn.Linear(in_features=num_proj_hidden, out_features=num_hidden)
+    def forward(self, x, edge_index):
+        return self.encoder(x, edge_index)
+
+    def projection(self, z: tlx.convert_to_tensor)->tlx.convert_to_tensor:
```

Comment:
What do you mean? The type of z is `tlx.convert_to_tensor`? Remember to correct the other similar error.